File: csc330/poundsPractice/Drink.java
public class Drink {

// the following will be class variables

   final int MAXIMUM_CAPACITY = 10;
   final double DEFAULT_DRINK_COST = 1.75;

   private int available, consumed;
   private int useDefaultCost, useSuppliedCost;
   private double drinkCost, moneyMade;
   private String name;

// Start defining constructors here...

// This constructor sets the name of the drink only.

   public Drink (String brandName) {
      name = brandName;
      consumed = 0;
      available = 0;
      moneyMade = 0.0;
      drinkCost = DEFAULT_DRINK_COST;
      useDefaultCost = 1;
      useSuppliedCost = 0;
   }

   public Drink (String brandName, double cost) {
      name = brandName;
      consumed = 0;
      available = 0;
      moneyMade = 0.0;
      drinkCost = cost;
      useDefaultCost = 0;
      useSuppliedCost = 1;
   }

// This is the basic vend method.  
   public void vend () {
      if ( available > 0 ) {
         available -= 1;
         consumed += 1;
         moneyMade += ( useDefaultCost * DEFAULT_DRINK_COST +
                        useSuppliedCost * drinkCost );
         System.out.println(name + " Puchased.");
         }
      else {
         System.out.println("************");
         System.out.println("* SOLD OUT *");
         System.out.println("************");
        }
     }


// Add a "setter" function called fillDrinks to set the number of
// drinks to their maximum capacity.
   public void refill () {
      available = MAXIMUM_CAPACITY;
      }


   public double profit () {
      return Math.round(moneyMade*100.0)/100.0;
      }


   public int drinksSold () {
      return consumed;
      }

   public String getDrinkName () {
      return name;
      }

   public int restockAmount () {
       return MAXIMUM_CAPACITY - available;
   }
}

File: csc330/poundsPractice/goPrac/Drink.go
package main

import (
        "fmt"
        "math"
       )

// Drink represents a beverage that can be purchased from a vending machine.
type Drink struct {
    // Class variables
    MaximumCapacity int
        DefaultDrinkCost float64

        // Instance variables
        Available int
        Consumed int
        UseDefaultCost int
        UseSuppliedCost int
        DrinkCost float64
        MoneyMade float64
        Name string
}

// NewDrink creates a new Drink object.
func NewDrink(name string) *Drink {
    return &Drink{
     Name : name,
     Consumed : 0,
     Available : 0,
     MoneyMade : 0.0,
     DrinkCost : 1.75,
     UseDefaultCost : 1,
     UseSuppliedCost : 0,
     MaximumCapacity : 10,
     DefaultDrinkCost : 1.75,
    }
}

// NewDrinkWithCost creates a new Drink object with a custom cost.
func NewDrinkWithCost(name string, cost float64) *Drink {
    return &Drink{
          Name: name,
          Consumed: 0,
          Available: 0,
          MoneyMade: 0.0,
          DrinkCost: cost,
          UseDefaultCost: 0,
          UseSuppliedCost: 1,
          MaximumCapacity: 10,
          DefaultDrinkCost: 1.75,
    }
}

// Vend dispenses a Drink from the vending machine.
func (d *Drink) Vend() {
    if d.Available > 0 {
        d.Available--
            d.Consumed++
            d.MoneyMade += (float64(d.UseDefaultCost) * d.DefaultDrinkCost +
                    float64(d.UseSuppliedCost) * d.DrinkCost)
            fmt.Println(d.Name + " purchased.")
    } else {
        fmt.Println("************")
        fmt.Println("* SOLD OUT *")
        fmt.Println("************")
    }
}

// Refill fills the vending machine with Drinks to their maximum capacity.
func (d *Drink) Refill() {
    d.Available = d.MaximumCapacity
}

// Profit calculates the profit made from selling Drinks.
func (d *Drink) Profit() float64 {
    return math.Round(d.MoneyMade*100.0) / 100.0
}

// DrinksSold returns the number of Drinks that have been sold.
func (d *Drink) DrinksSold() int {
    return d.Consumed
}

// GetDrinkName returns the name of the Drink.
func (d *Drink) GetDrinkName() string {
    return d.Name
}

// RestockAmount returns the number of Drinks that need to be restocked.
func (d *Drink) RestockAmount() int {
    return d.MaximumCapacity - d.Available
}

